tests/tron.py

```diff
@@ -0,0 +1,346 @@
+
+        data = bytearray.fromhex(f"05{path}")
+        while len(tx) > 0:
+            # get next message field
+            newpos = self.get_next_length(tx)
+            assert (newpos < MAX_APDU_LEN)
+            if (len(data) + newpos) > MAX_APDU_LEN:
+                # add chunk
+                to_send.append(data.hex())
+                data = bytearray()
+                continue
+            # append to data
+            data.extend(tx[:newpos])
+            tx = tx[newpos:]
+        # append last
+        to_send.append(data.hex())
+        token_pos = len(to_send)
+        to_send.extend(signatures)
+
+        if len(to_send) == 1:
+            start_bytes.append(P1.SIGN)
+        else:
+            start_bytes.append(P1.FIRST)
+            for i in range(1, len(to_send) - 1):
+                if (i >= token_pos):
+                    start_bytes.append(P1.TRC10_NAME | P1.FIRST
+                                       | i - token_pos)
+                else:
+                    start_bytes.append(P1.MORE)
+
+            if not (signatures == None) and len(signatures) > 0:
+                start_bytes.append(P1.TRC10_NAME
+                                   | InsType.SIGN_PERSONAL_MESSAGE
+                                   | len(signatures) - 1)
+            else:
+                start_bytes.append(P1.LAST)
+
+        for i in range(len(to_send)):
+            pack = self.apduMessage(InsType.SIGN, start_bytes[i], 0x00,
+                                    to_send[i])
+            if i < len(to_send) - 1:
+
+                ret = self._exchange_raw(payload=pack)
+                if not (ret.status == Errors.OK):
+                    raise ValueError(
+                        "Something went wrong while sending the APDU.")
+
+            else:
+                with self._client.exchange_async_raw(pack) as r:
+                    if navigate:
+                        if self._firmware.device == "stax":
+                            sleep(1.5)
+                            self._navigator.navigate_until_text_and_compare(
+                                NavIns(NavInsID.TOUCH, (200, 545)),
+                                [NavIns(NavInsID.USE_CASE_REVIEW_CONFIRM)],
+                                text,
+                                Path(__file__).parent.resolve(),
+                                snappath,
+                                screen_change_after_last_instruction=False)
+                        else:
+                            self._navigator.navigate_until_text_and_compare(
+                                NavIns(NavInsID.RIGHT_CLICK),
+                                [NavIns(NavInsID.BOTH_CLICK)],
+                                text,
+                                Path(__file__).parent.resolve(),
+                                snappath,
+                                screen_change_after_last_instruction=False)
+                    else:
+                        pass
+                    sleep(0.5)
```

Comment:
```suggestion
                if navigate:
                    self.exchange_async_and_navigate(pack, snappath, text)
                else:
                    self._client.exchange_raw(pack):
```
I think that should do the same job?

Reply:
Yes, good point